### PR TITLE
refactor(ff): re-builds fairs index page

### PR DESCRIPTION
**Pull request by dzucconi**
### ToDo

- [x] View more button
- [x] Specs
- [x] Tighten up mobile layout (new tabs)
- [x] [State for no current events](http://static.damonzucconi.com/_capture/F3rn6U1ipBGb.png)

### Clean up
- [ ] Analytics?
- [ ] Remove old fairs app (both desktop and mobile)

-----

### CODE REVIEW

**Comment by joeyAghion** [comment]
Nice!

Needs another [ ] (or separate PR) to remove the old implementation(s), right?

**Comment by dzucconi** [comment]
This should be ready and will open this off draft once https://github.com/artsy/metaphysics/pull/2919 is deployed.

File: data/schema.graphql [hunk 1]
```diff
@@ -11209,6 +11227,24 @@ type Viewer {
...
+    hasListing: Boolean
+
+    # Only return fairs matching specified IDs. Accepts list of IDs.
+    ids: [String]
+    last: Int
+    near: Near
+    sort: FairSorts
+    status: EventStatus
+  ): FairConnection
+
```
Comment (dzucconi) [inline data/schema.graphql:11247]: This commit will go away; was just to work against staging.

**Comment by mzikherman** [comment]
Aw yea! 🔥 😎

**Comment by dzucconi** [comment]
Will merge this later today unless you have any comments @damassi — it won't be linked anywhere yet so just so we can Q/A against the existing page before swapping it over.

File: src/v2/Apps/Fairs/Components/FairsFairBanner.tsx [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    fair.image &&
+    (fair.bannerSize === "x-large" ? fair.image.large : fair.image.small)
+
+  const icon = fair?.profile?.icon?.resized
+
+  return (
+    <Box {...rest}>
+      <RouterLink
+        to={fair.href}
+        style={{ display: "block", textDecoration: "none" }}
```
Comment (damassi) [inline src/v2/Apps/Fairs/Components/FairsFairBanner.tsx:40]: (general) we add `textDecoration: none` everywhere this is used -- we should probably reverse this and opt out by default and opt if needed.

File: src/v2/Apps/Fairs/Components/FairsFairRow.tsx [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        py={2}
+        px={[0, 2]}
+        {...rest}
+      >
+        <Box
+          display="flex"
+          flex="1"
+          alignItems={["flex-start", "center"]}
+          flexDirection={["column", "row"]}
+        >
```
Comment (damassi) [inline src/v2/Apps/Fairs/Components/FairsFairRow.tsx:42]: (nonblocking) No longer use `<Flex>`? IMO it reads a bit better at a glance, and is what most people use.

File: src/v2/Apps/Fairs/Components/FairsFairBanner.tsx (hunk 2 above)
Comment (dzucconi) [reply src/v2/Apps/Fairs/Components/FairsFairBanner.tsx:40]: I concur — there's a `noUnderline` prop but it doesn't work correctly if you want to also add display options. We should revisit this component a bit and mixin some styled-system functions

File: src/v2/Apps/Fairs/Components/FairsPromoCarousel.tsx [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+          height="100%"
+          cursor="e-resize"
+          onClick={event => {
+            setInteracted(true)
+            onClick(event)
+          }}
+          {...rest}
+        />
+      )}
+      Previous={() => <></>}
```
Comment (damassi) [inline src/v2/Apps/Fairs/Components/FairsPromoCarousel.tsx:53]: ?

Comment (dzucconi) [reply src/v2/Apps/Fairs/Components/FairsPromoCarousel.tsx:53]: Basically just nulls the previous arrow. This is a very weird carousel of course, and it just has a big next button overlaying the whole thing.

File: src/v2/Apps/Fairs/Components/FairsFairRow.tsx (hunk 3 above)
Comment (dzucconi) [reply src/v2/Apps/Fairs/Components/FairsFairRow.tsx:42]: I personally find it more confusing; having to switch between two components for a single display option.

File: src/v2/Apps/Fairs/Routes/FairsIndex.tsx [hunk 5]
```diff
@@ -0,0 +1,355 @@
...
+        <ResponsiveBox aspectWidth={16} aspectHeight={9} maxWidth="100%">
+          <Text
+            as="h1"
+            variant="largeTitle"
+            color="white100"
+            mx={-2}
+            height="100%"
+            backgroundImage="url('https://files.artsy.net/images/fairs-header-img.jpg')"
+            backgroundSize="cover"
+            backgroundPosition="center center"
```
Comment (damassi) [inline src/v2/Apps/Fairs/Routes/FairsIndex.tsx:74]: Ooh this is an interesting approach to ths

File: src/v2/Apps/Fairs/Components/FairsPromoCarousel.tsx (hunk 4 above)
Comment (dzucconi) [reply src/v2/Apps/Fairs/Components/FairsPromoCarousel.tsx:53]: TBH, we'll probably just drop all this stuff once we have new templates to work with; though it was an interesting exercise to try to recreate it with the new carousel.

File: src/v2/Apps/Fairs/fairsRoutes.tsx [hunk 6]
```diff
@@ -0,0 +1,41 @@
+/* eslint-disable sort-keys-fix/sort-keys-fix */
```
Comment (damassi) [inline src/v2/Apps/Fairs/fairsRoutes.tsx:1]: This can be removed

**Review by damassi — APPROVED** [review]
Looks great @dzucconi - only a couple non-blocking Qs 👍

File: src/v2/Apps/Fairs/Components/FairsFairRow.tsx (hunk 3 above)
Comment (damassi) [reply src/v2/Apps/Fairs/Components/FairsFairRow.tsx:42]: This way is totally cool, but its def unconventional across our UI code in Force / Eigen.

**Comment by dzucconi** [comment]
Thanks for the feedback @damassi — addressed & merging.

**Pull request merged**